tests/backend/checkmach/dune

```diff
@@ -1,18 +1,94 @@
 (rule
  (alias   runtest)
  (enabled_if (= %{context_name} "main"))
- (targets t.o t.cmx s.o s.cmx output.corrected t.cmx.output)
  (deps s.ml t.ml)
+ (action (run %{bin:ocamlopt.opt} %{deps} -c -alloc-check -O3)))
+
+(rule
+ (alias   runtest)
+ (enabled_if (= %{context_name} "main"))
+ (deps test_assume.ml)
+ (action (run %{bin:ocamlopt.opt} %{deps} -c -alloc-check -O3)))
+
+(rule
+ (alias   runtest)
+ (enabled_if (and (= %{context_name} "main")
+                   %{ocaml-config:flambda}
+                  ;; what we really want to say but dune doesn't know about flambda2:
+                  ;; (or %{ocaml-config:flambda} %{ocaml-config:flambda2})
+                  ))
+ (deps test_flambda.ml)
+ (action (run %{bin:ocamlopt.opt} %{deps} -c -alloc-check -O3)))
+
+(rule
+ (enabled_if (= %{context_name} "main"))
+ (targets fail1.output.corrected)
+ (deps fail1.ml)
  (action
-    (progn
-     (run %{bin:ocamlopt.opt} %{deps} -c -alloc-check -O3)
-     (with-stdout-to t.cmx.output (run %{bin:ocamlobjinfo} t.cmx))
-
-     ; CR gyorsh: count the lines in the relevant section of the output of ocamlobjinfo,
-     ; instead of printing the function symbols whose names may differ
-     ; depending on configuration.
-     ; This is still flaky and temporary, until @assert annotations can be checked.
-     (with-stdout-to output.corrected
-       (with-stdin-from t.cmx.output
-         (bash "sed -n '/Functions with neither allocations nor indirect calls:/,$p' | grep -v camlT__entry | wc -l | sed 's/^ *//g' ")))
-     (diff output output.corrected))))
+   (with-outputs-to fail1.output.corrected
+    (pipe-outputs
+    (with-accepted-exit-codes 2
+     (run %{bin:ocamlopt.opt} %{deps} -color never -error-style short -c -alloc-check -O3))
```

Comment:
I think this "HIDE_NAME" token
is too wide: we should probably
strip the stamp but keep the
function name. We would be
making the output easier to read
and test a tiny bit more.

Reply:
Ah, yes, I forgot about the different
mangling scheme flambda2 is using.